src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs

```diff
@@ -250,172 +180,183 @@ private void DeleteSelectedButton_Click(object sender, RoutedEventArgs e)
-                        graphic.Symbol = _multiPointSymbol;
-                        break;
-
-                    case GeometryType.Polyline:
-                        graphic.Symbol = _lineSymbol;
-                        break;
-
-                    case GeometryType.Polygon:
-                        graphic.Symbol = _fillSymbol;
-                        break;
-                }
-
-                _graphicsOverlay.Graphics.Add(graphic);
+                // Create a new graphic based on the geometry and add it to the graphics overlay.
+                _graphicsOverlay.Graphics.Add(new Graphic(geometry, GetSymbol(geometry.GeometryType)));
             }
 
-            // Allow the creation of new graphics.
-            EnableGeometryButtons();
-            ToolComboBox.IsEnabled = true;
+            ResetFromEditingSession();
         }
 
         // Stops the geometry editor without saving the geometry stored within.
         private void DiscardButton_Click(object sender, RoutedEventArgs e)
         {
-            if (_selectedGraphic != null)
-            {
-                // Make visisble the pre-existing graphic when editing.
-                _selectedGraphic.IsVisible = true;
-                _selectedGraphic.IsSelected = false;
-                _selectedGraphic = null;
-
-                _geometryEditor.ClearSelection();
-            }
-
             _geometryEditor.Stop();
-
-            // Allow for the creation of new graphics.
-            EnableGeometryButtons();
-            ToolComboBox.IsEnabled = true;
+            ResetFromEditingSession();
         }
 
         // Removes all graphics from the graphics overlay.
         private void DeleteAllButton_Click(object sender, RoutedEventArgs e)
         {
             _graphicsOverlay.Graphics.Clear();
+        }
 
-            // Disable the delete all button as there are no graphics to delete.
-            DeleteAllButton.IsEnabled = false;
-            _geometryCreated = false;
+        private async void MyMapView_GeoViewTapped(object sender, GeoViewInputEventArgs e)
+        {
+            // Return immediately when in an editing session.
+            if (_geometryEditor.IsStarted) return;
 
-            // Allow for the creation of new graphics.
-            EnableGeometryButtons();
-            ToolComboBox.IsEnabled = true;
+            try
+            {
+                // Identify graphics in the graphics overlay using the mouse point.
+                IReadOnlyList<IdentifyGraphicsOverlayResult> results = await MyMapView.IdentifyGraphicsOverlaysAsync(e.Position, 5, false);
+
+                // Try to get the first graphic from the first result.
+                _selectedGraphic = results.FirstOrDefault()?.Graphics?.FirstOrDefault();
+            }
+            catch (Exception ex)
+            {
+                // Report exceptions.
+                MessageBox.Show("Editing error: " + ex.Message, ex.GetType().Name, MessageBoxButton.OK, MessageBoxImage.Error);
```

Comment:
I don't know if there's a precedence in the sample already on how message is displayed but I don't think this will be an "editing error" but an error in the identify operation.

Reply:
Will make it display as "error"